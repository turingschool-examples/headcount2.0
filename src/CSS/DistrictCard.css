@font-face {
  font-family: anon;
  src: url(../Fonts/AnonymousPro-Regular.ttf);
}

.avg-low {
  font-family: anon;
  color: rgb(169, 77, 77);
}

.avg-high {
  font-family: anon;
  color: green;
}

.year {
  font-family: anon;
  color: black;
}

.stats-show {
  text-align: left;
  line-height: 5px;
  padding-left: 20px;
}

.stats-hide {
  display: none;
}

.year-header-show,
.avg-header-show {
  font-size: 1.3rem;
  margin: 0 auto;
  display: inline-block;
  margin: 0 2rem;
}

.year-header-hide,
.avg-header-hide {
  display: none;
}

.district-card {
  cursor: pointer;
  background-color: rgb(228, 221, 221);
  border: 2px solid rgb(155, 153, 153);
  text-align: center;
  box-shadow: 1.5px 1.5px 1.5px 1.5px #ccc;
  border-radius: 4px;
  animation: bazinga 1s ease-in;
}

.district-card-clicked1 {
  position: absolute;
  cursor: pointer;
  border-radius: 4px;
  border: 2px solid rgb(135, 111, 111);
  text-align: center;
  box-shadow: 1.5px 1.5px 1.5px 1.5px #ccc;
  top: 10rem;
  left: 3rem;
  animation: move 0.5s linear forwards, bazinga 1s ease-in;
}

.district-card-clicked2 {
  position: absolute;
  cursor: pointer;
  border-radius: 4px;
  border: 2px solid rgb(135, 111, 111);
  text-align: center;
  box-shadow: 1.5px 1.5px 1.5px 1.5px #ccc;
  top: 35rem;
  left: 3rem;
  animation: move 0.5s linear forwards, bazinga 1s ease-in;
}

.district-card:hover {
  background-color: white;
  box-shadow: 1.5px 1.5px 1.5px 1.5px rgb(181, 168, 168);
}

.district-card-clicked1:hover {
  background-color: rgb(228, 221, 221);
  box-shadow: 1.5px 1.5px 1.5px 1.5px rgb(140, 135, 135);
}

.district-card-clicked2:hover {
  background-color: rgb(228, 221, 221);
  box-shadow: 1.5px 1.5px 1.5px 1.5px rgb(148, 140, 140);
}

.district-location {
  font-size: 1.3rem;
  width: 14rem;
  text-align: center;
  margin: 1rem auto;
  align-self: center;
}

.hide-msg {
  display: none;
}

@keyframes bazinga {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes move {
  0% {
    left: 0%;
    transform: rotate3d(0, 0, 0, 0);
  }
}
