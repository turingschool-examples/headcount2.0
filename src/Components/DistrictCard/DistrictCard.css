.highStats {
  color: #7097B2;
}

.lowStats {
  color: #B81F38;
}

.card {
  border-radius: 18px;
  border: solid #4F4C47 2px;
  color: #4F4C47;
  display: inline-block;
  margin: 1rem;
  padding: .5rem;
  width: 300px;
}

.selected {
  background-color: #F7F5F1;
  border: solid #21201E 4px;
}

.card:hover {
  border: solid #4F4C47 4px;
}

.district {
  font-family: avenir, sans-serif;
  font-size: 1.5em;
  text-align: center;
}

.stats {
  font-family: avenir, sans-serif;
  font-size: 1.25em;
  padding-left: 5rem;
}