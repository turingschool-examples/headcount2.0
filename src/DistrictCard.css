.districtCard {
  box-shadow: 1em;
  border-radius: 15px;
  background-color: white;
  display: flex;
  flex-direction: column;
  width: 15rem;
  height: auto;
  margin: 0 0 5rem 0;
  padding-bottom: 1rem;
  text-align: center
}

.districtLocation {
  font-size: 1rem
}

.districtStats {
  list-style-type: none;
  text-align: center;
  margin: 0;
  padding: 0
}

.statGreater {
  color: blue
}

.statLesser {
  color: red
}