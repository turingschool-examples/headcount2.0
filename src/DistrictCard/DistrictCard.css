.DistrictCard {
	display: flex;
	align-items: center;
	flex-direction: column;
	justify-content: center;
	background-color: #FFF;
	text-align: center;
	width: 15rem;
	height: 26rem;
	margin: 0 auto 0.5rem;
	padding: 0 0 1rem;
	box-shadow: 0px 1px 1px 2px #F5F6F7;
	border-radius: 10px;
	font-family: 'Open Sans', sans-serif;
}

.card-location {
	text-align: center;
	overflow-wrap: break-word;
	margin: 1.5rem 1.5rem 0.75rem;
	font-family: 'Arvo', serif;
	font-size: 1.25rem;
}

/*.card-stats {
	margin: 0 auto 1.5rem;
}
*/
.greater-than-point-5 {
	color: #01411b;
	background-color: #F5F6F7;
	margin: 0.2rem auto;
	font-weight: 300;
	padding: 0 0.75rem;
	border-radius: 10px;
}

.less-than-point-5 {
	color: #FFF;
	background-color: #008b38;
	margin: 0.2rem auto;
	font-weight: 300;
	padding: 0 0.75rem;
	border-radius: 10px;
}


