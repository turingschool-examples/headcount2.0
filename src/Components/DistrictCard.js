import React from "react";
import "../CSS/DistrictCard.css";
import PropTypes from "prop-types";

export const DistrictCard = ({
  location,
  stats,
  clicked,
  selectCard,
  counter
}) => {
  return (
    <div
      onClick={() => selectCard(location)}
      className={clicked ? "district-card-clicked" + counter : "district-card"}
    >
      <h1 className="district-location">{location}</h1>
      <header className={clicked ? "year-header-show" : "year-header-hide"}>
        Year
      </header>
      <header className={clicked ? "avg-header-show" : "avg-header-hide"}>
        Avg
      </header>
      {Object.keys(stats).map((year, index) => {
        const avg = stats[year];
        return (
          <section
            className={clicked ? "stats-show" : "stats-hide"}
            key={index}
          >
            <p className="year">
              {year}
              ......
              <span className={avg < 0.5 ? "avg-low" : "avg-high"}>{avg}</span>
            </p>
          </section>
        );
      })}
      <p className={clicked ? "show-msg" : "hide-msg"}>Click to Remove</p>
    </div>
  );
};

const { string, shape, number, bool, func } = PropTypes;

DistrictCard.propTypes = {
  location: string,
  stats: shape({ year: number }),
  clicked: bool,
  selectCard: func,
  counter: number
};
