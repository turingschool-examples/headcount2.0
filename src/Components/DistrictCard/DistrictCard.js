import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';


const DistrictCard = ({district , selectDistrict}) => {
  const years = Object.keys(district.stats)

  const determineColor = (stats) => {
    return stats > 0.5 ? "highStats" : "lowStats"
  }

  const createStats = years.map((year, index) => {
    const stats = district.stats[year]
    return (
      <li className={determineColor(stats)} key={index}>
        {[year]}: {stats}
      </li>
      )
  })

  const handleClick = (location) => {
    selectDistrict(location)
  }

  return (
    <section 
      className={`card ${district.selected ? 'selected' : ''}`} 
      id={district.key}
      key={district.key}
      onClick={() => {handleClick(district.location)}}
    > 
      <h2 className="district">District Name: {district.location}</h2>
      <ul className="stats">
        {createStats}
      </ul>
    </section>
  )
}

DistrictCard.propTypes = { 
  district: PropTypes.object
}



export default DistrictCard;