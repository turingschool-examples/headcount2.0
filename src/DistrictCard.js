import React from "react";
import "./CSS/DistrictCard.css";

export const DistrictCard = ({ location, stats, selectCard }) => {
  return (
    <div className="district-card">
      <h1 className="district-location">{location}</h1>
      {Object.keys(stats).map((year, i) => {
        const avg = stats[year];
        return (
          <section
            className={avg < 0.5 ? "year-red" : "year-black"}
            key={i}
            onClick={e => selectCard(location)}
          >
            {year} :
            <span className={avg < 0.5 ? "year-red" : "year-black"}>{avg}</span>
          </section>
        );
      })}
    </div>
  );
};
