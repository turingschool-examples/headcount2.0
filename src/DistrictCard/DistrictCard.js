import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';

const DistrictCard = ({ location, stats, district, compareDistrict }) => {

	const statsKeys = Object.keys(stats)
	const schoolData = statsKeys.map((stat) => {
		// console.log(stat)
		return <p 
			key={Math.random()}
			className={(stats[stat] > 0.5) ? 'greater-than-point-5' : 'less-than-point-5'}>
			{stat}: {stats[stat]} 
		</p>
	})
	
if (compareDistrict) {
		return (
		<div className='DistrictCard' onClick={ () => compareDistrict(district) }>
			<h3 className='card-location'>{location}</h3>
			{schoolData}
		</div>
	)
	} else {
		return (
			<div className='DistrictCard'>
				<h3 className='card-location'>{location}</h3>
				{schoolData}
			</div>
		)
	}

}

DistrictCard.proptypes = {
	location: PropTypes.string.isRequired,
	stats: PropTypes.object.isRequired,
	district: PropTypes.object.isRequired,
	compareDistrict: PropTypes.func
}

export default DistrictCard;
