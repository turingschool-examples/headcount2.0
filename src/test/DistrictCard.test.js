import React from "react";
import { DistrictCard } from "../Components/DistrictCard";

describe("DistrictCard", () => {
  let shallowWrap;
  let mockLocation = {
    clicked: true,
    location: "Colorado",
    stats: {
      "2004": 0.24,
      "2005": 0.278,
      "2006": 0.337,
      "2007": 0.395,
      "2008": 0.536,
      "2009": 0.598,
      "2010": 0.64,
      "2011": 0.672,
      "2012": 0.695,
      "2013": 0.703,
      "2014": 0.741
    }
  };

  beforeEach(() => {
    shallowWrap = shallow(
      <DistrictCard
        location={mockLocation.location}
        stats={mockLocation.stats}
        selectCard={jest.fn()}
      />
    );
  });

  it("should exist", () => {
    expect(shallowWrap).toBeDefined();
  });

  it("should handle click of individual card", () => {
    let mockfn = jest.fn();
    let mockWrap = shallow(
      <DistrictCard
        location={mockLocation.location}
        stats={mockLocation.stats}
        selectCard={mockfn}
        index={1}
      />
    );
    let sectionOne = mockWrap.find("div").first();

    sectionOne.simulate("click");

    expect(mockfn).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    shallowWrap = shallow(
      <DistrictCard location={""} stats={{}} selectCard={jest.fn()} />
    );
    expect(shallowWrap).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    expect(shallowWrap).toMatchSnapshot();
  });
});
