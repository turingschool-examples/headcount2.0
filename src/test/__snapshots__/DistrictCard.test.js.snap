// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictCard should match snapshot 1`] = `
<div
  className="district-card"
  onClick={[Function]}
>
  <h1
    className="district-location"
  />
  <header
    className="year-header-hide"
  >
    Year
  </header>
  <header
    className="avg-header-hide"
  >
    Avg
  </header>
  <p
    className="hide-msg"
  >
    Click to Remove
  </p>
</div>
`;

exports[`DistrictCard should match snapshot 2`] = `
<div
  className="district-card"
  onClick={[Function]}
>
  <h1
    className="district-location"
  >
    Colorado
  </h1>
  <header
    className="year-header-hide"
  >
    Year
  </header>
  <header
    className="avg-header-hide"
  >
    Avg
  </header>
  <section
    className="stats-hide"
    key="0"
  >
    <p
      className="year"
    >
      2004
      ......
      <span
        className="avg-low"
      >
        0.24
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="1"
  >
    <p
      className="year"
    >
      2005
      ......
      <span
        className="avg-low"
      >
        0.278
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="2"
  >
    <p
      className="year"
    >
      2006
      ......
      <span
        className="avg-low"
      >
        0.337
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="3"
  >
    <p
      className="year"
    >
      2007
      ......
      <span
        className="avg-low"
      >
        0.395
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="4"
  >
    <p
      className="year"
    >
      2008
      ......
      <span
        className="avg-high"
      >
        0.536
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="5"
  >
    <p
      className="year"
    >
      2009
      ......
      <span
        className="avg-high"
      >
        0.598
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="6"
  >
    <p
      className="year"
    >
      2010
      ......
      <span
        className="avg-high"
      >
        0.64
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="7"
  >
    <p
      className="year"
    >
      2011
      ......
      <span
        className="avg-high"
      >
        0.672
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="8"
  >
    <p
      className="year"
    >
      2012
      ......
      <span
        className="avg-high"
      >
        0.695
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="9"
  >
    <p
      className="year"
    >
      2013
      ......
      <span
        className="avg-high"
      >
        0.703
      </span>
    </p>
  </section>
  <section
    className="stats-hide"
    key="10"
  >
    <p
      className="year"
    >
      2014
      ......
      <span
        className="avg-high"
      >
        0.741
      </span>
    </p>
  </section>
  <p
    className="hide-msg"
  >
    Click to Remove
  </p>
</div>
`;
